src/semgrep_agent/formatter.py

```diff
@@ -68,9 +84,69 @@ def _print_error_message(violation: Finding) -> str:
     return f"{BOLD}{rule}{END}"
 
 
-def dump(findings: Set[Finding]) -> None:
+OSC_8 = "\x1b]8;;"
+BEL = "\x07"
+
+
+def is_child_process_of(pattern: Pattern) -> bool:
+    """
+    Returns true iff this process is a child process of a process whose name matches pattern
+    """
+    me = psutil.Process()
+    parents = me.parents()
+    matches = iter(0 for p in parents if pattern.search(p.name()))
+    return next(matches, None) is not None
+
+
+LINK_PRINTER_PATTERN = re.compile("(iterm2|gnome-terminal)", re.IGNORECASE)
+LINK_WIDTH = 2 * len(OSC_8) + 2 * len(BEL)
+DO_PRINT_LINKS = is_child_process_of(LINK_PRINTER_PATTERN)
+
+
+def render_link(
+    text: str,
+    href: Optional[str],
```

Comment:
```suggestion
    href: Optional[str],
    *,
```

I'd force setting the param name for everything below, too easy to mix up the order of boolean flags.

Reply:
Good idea and willdo (By the way -- this function is copied straight from the old bento code :))